pgjdbc/src/main/java/org/postgresql/jdbc/BooleanTypeUtil.java

```diff
@@ -22,6 +25,28 @@
 
   private static final Logger LOGGER = Logger.getLogger(BooleanTypeUtil.class.getName());
 
+  /**
+   * Maps String values to Boolean constants. By using {@link String#CASE_INSENSITIVE_ORDER}
+   * {@link Map#get(Object)} is case-insensitive.
+   */
+  private static final SortedMap<String, Boolean> STRING_TO_BOOL = new TreeMap<String, Boolean>(
```

Comment:
I threw together a quick jmh benchmark. There ends up not being much difference between these 2 approaches on average. There are specific values which are faster and slower for each.
In playing with various options, I found an approach that cuts the time roughly in half (on average), which is using a concurrent hash map assuming lower case and add in the handling for alternative case when encountered at runtime.